src/experimental/tools/download2go/index.ts

```diff
@@ -0,0 +1,415 @@
+            }
+            const { progress, manifest, audio, video, text, size } = download;
+            if (manifest === null) {
+              return;
+            }
+            const { metaData, transport, duration } = storedManifest;
+            db.put("manifests", {
+              contentID,
+              transport,
+              manifest,
+              builder: { video, audio, text },
+              progress,
+              size,
+              duration,
+              metaData,
+            }).then(() =>
+              this.trigger("insertDB", {
+                action: "store",
+                contentID,
+                progress: progress.percentage,
+              })
+            ).catch((error: Error) => {
+              this.trigger("error", {
+                action: "resume-downloader",
+                error,
+                contentID,
+              });
+            });
+          },
+          (error: Error) =>
+            this.trigger("error", {
+              action: "resume-downloader",
+              error,
+              contentID,
+            })
+        );
+      this.activeDownloads[contentID] = resumeDownloadSub;
+    } catch (err) {
+      const error = err as Error;
+      this.trigger("error", { action: "resume-downloader", error, contentID });
+    }
+  }
+
+  /**
+   * Pause a download already started earlier.
+   * @param {string} contentID
+   * @returns {void}
+   */
+  pause(contentID: string): void {
+    try {
+      checkForPauseAMovie(contentID);
+      const activeDownloads = this.activeDownloads;
+      const activePauses = this.activePauses;
+      if (activeDownloads[contentID] == null && activePauses[contentID] == null) {
+        throw new ValidationArgsError(`Invalid contentID given: ${contentID}`);
+      }
+      activePauses[contentID].next();
+      activePauses[contentID].complete();
+      activeDownloads[contentID].unsubscribe();
+      delete activeDownloads[contentID];
+      delete activePauses[contentID];
+    } catch (e) {
+      const error = e as Error | undefined;
+      this.trigger("error", {
+        action: "pause",
+        contentID,
+        error: error !== undefined ? error : new Error("A Unexpected error happened"),
+      });
+    }
+  }
+
+  /**
+   * Get all the downloaded entry (manifest) partially or fully downloaded.
+   * @returns {Promise.<IStoredManifest[]|void>}
+   */
+  getAllOfflineContent(
+    limit?: number
+  ): Promise<IStoredManifest[] | undefined> | void {
+    try {
+      if (this.db == null) {
+        throw new Error("The IndexDB database has not been created!");
+      }
+      // TODO: return formatted content ?
+      return this.db.getAll("manifests", undefined, limit);
+    } catch (e) {
+      const error = e as Error;
+      this.trigger("error", {
+        action: "getAllDownloadedMovies",
+        error: new IndexDBError(error.message),
+      });
+    }
+  }
+
+  /**
+   * Get a singleMovie ready to be played by the rx-player,
```

Comment:
We don't download "Movie", we download "contents". We could for example download an audio content or a content with multiple "movies" in it. Or even a TV show, which is not a "movie".
I prefer we remove any mention of the word "movie" in here.

Reply:
Absolutely agree, this is 'development' naming. I will fix that for sure!